### DISCUSSION TITLE
Setting up Beebotte MQTT

### DISCUSSION THREAD

**Discussion by i-g-g-y** [jomjol/AI-on-the-edge-device#2660] [Q&A]
Can you help?
![image](https://github.com/jomjol/AI-on-the-edge-device/assets/10361323/2c1556f6-8d6c-445c-bf78-4f7229fe1199)

This does not seem to work:
![image](https://github.com/jomjol/AI-on-the-edge-device/assets/10361323/3a553d98-55c4-45f9-96a7-0f3091bbe877)

**Comment by caco3** [comment]
You will need to provide more information!
Is there an error shown?
Does the broker receive the data?
what do other MQTT clients see? I usually use http://mqtt-explorer.com/ for this.

**Reply by i-g-g-y** [reply]
![image](https://github.com/jomjol/AI-on-the-edge-device/assets/10361323/c20212bd-a8be-4865-8c27-f0073ad34ae1)

**Reply by caco3** [reply]
Are the credentials correct?
Does it really work using a token as username?

Does the server expect any encryption? Somebody else just is adding TLS and sais that some cloud services require it, see https://github.com/jomjol/AI-on-the-edge-device/issues/2625

**Reply by i-g-g-y** [reply]
Honestly I do not know. When writing c code, I used token as authentication, no password, port 1883, so everything should work. It is highly specific, so more of a question to public..

![image](https://github.com/jomjol/AI-on-the-edge-device/assets/10361323/1c64b5f5-5c5c-4810-abdf-558478d26da8)

**Reply by caco3** [reply]
To my experience, one usually can use a token instead of user/password. How ever the token gets transmitted in a different ways, you can't just use it as the suer name or password!

Examples with `curl`:
```bash
curl -u "login:password" ...
curl -H "Authorization: Bearer {token}" ...
```

Is it possible you can use a username/password?

**Reply by caco3** [reply]
Maybe you have to prepend `token:` before the username?
See https://gist.github.com/bachwehbi/a938f4bc8fdde5e6a732#file-bbt_mqtt-js-L13

**Reply by i-g-g-y** [reply]
Give me a few minutes, I will give it a shot. Was not needed in c program, but maybe this was hidden deep in library..

**Reply by i-g-g-y** [reply]
No dice..
![image](https://github.com/jomjol/AI-on-the-edge-device/assets/10361323/f5854785-9db8-4ed5-a7f2-94d0f6ad336e)
